### PR TITLE
Homework 2 task 3

### PR DESCRIPTION
**Pull request by ivanzuev78**
Homework 2 task 3 done

### CODE REVIEW

File: HomeWork_2/hw3.py [hunk 1]
```diff
@@ -18,4 +18,15 @@
 
 
 def combinations(*args: List[Any]) -> List[List]:
-    ...
+
+    list_to_return = []
+
+    for i in args[0]:
+        if len(args) == 1:
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw3.py:25]: It will be more convenient to have this check before cycle.

File: HomeWork_2/hw3.py [hunk 2]
```diff
@@ -18,4 +18,15 @@
...
+            list_to_return.append([i])
+
+        else:
+            for end_of_array in combinations(*args[1:]):
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw3.py:29]: There is a nice name in function programming for that:  tail

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_2/hw3.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_2/hw3.py:25]: Yes, I agree.

File: HomeWork_2/hw3.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork_2/hw3.py:29]: I'll memorize it))

File: HomeWork_2/hw3.py [hunk 3]
```diff
@@ -18,4 +18,16 @@
 
 
 def combinations(*args: List[Any]) -> List[List]:
-    ...
+
+    list_to_return = []
+
+    if len(args) == 1:
+        for i in args[0]:
```
Comment (Cjkjvfnby) [inline HomeWork_2/hw3.py:25]: Can you change this for- loop to list comprehension or geenrator expression?

File: HomeWork_2/test_hw3.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from HomeWork_2.hw3 import combinations
+
+
+def test_combinations_example():
```
Comment (Cjkjvfnby) [inline HomeWork_2/test_hw3.py:4]: Have you covered all cases that can be in combinations?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_2/test_hw3.py (hunk 4 above)
Comment (ivanzuev78) [reply HomeWork_2/test_hw3.py:4]: Added 2 more tests.

File: HomeWork_2/hw3.py (hunk 3 above)
Comment (ivanzuev78) [reply HomeWork_2/hw3.py]: Of course I can.

File: HomeWork_2/test_hw3.py (hunk 4 above)
Comment (Cjkjvfnby) [reply HomeWork_2/test_hw3.py:4]: Need more. Check your implementation for ideas.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_2/test_hw3.py (hunk 4 above)
Comment (ivanzuev78) [reply HomeWork_2/test_hw3.py:4]: May be something like that?

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**